File: app/src/main/java/com/imagegallery/list/service/ImageService.java
package com.imagegallery.list.service;

import com.imagegallery.model.PhotoSearchResult;

import io.reactivex.Single;

public interface ImageService {

    int DATE_TAKEN_DESC_SORT_TYPE = 0;
    int DATE_PUBLISHED_DESC_SORT_TYPE = 1;

    Single<PhotoSearchResult> listPhotos();

    Single<PhotoSearchResult> listPhotos(int sortType);

    Single<PhotoSearchResult> searchPhotos(String query);

    Single<PhotoSearchResult> searchPhotos(String query, int sortType);

}

File: app/src/main/java/com/jamesgoodwin/flickrclient/list/service/ImageService.kt
package com.jamesgoodwin.flickrclient.list.service

import com.jamesgoodwin.flickrclient.model.PhotoSearchResult

import io.reactivex.Single

interface ImageService {

    fun listPhotos(): Single<PhotoSearchResult>

    fun searchPhotos(query: String): Single<PhotoSearchResult>

}